### PR TITLE
refactor(hummock): remove inf for key-range

### PR DESCRIPTION
**Pull request by Little-Wallace**
Signed-off-by: Little-Wallace <bupt2013211450@gmail.com>

I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

In fact, we already use `Bytes::empty()` to represent an unbounded (-inf or inf). So the inf flag is not necessary. 

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Documentation

If your pull request contains user-facing changes, please specify the types of the changes, and create a release note. Otherwise, please feel free to remove this section.

### Types of user-facing changes

Please keep the types that apply to your changes, and remove those that do not apply.

* Installation and deployment 
* Connector (sources & sinks)
* SQL commands, functions, and operators
* RisingWave cluster configuration changes
* Other (please specify in the release note below)

### Release note

Please create a release note for your changes. In the release note, focus on the impact on users, and mention the environment or conditions where the impact may occur.

### CODE REVIEW

**Comment by zwang28** [comment]
+1, it's not always correctly set currently.

File: src/storage/hummock_sdk/src/key_range.rs [hunk 1]
```diff
@@ -22,26 +22,17 @@ use super::version_cmp::VersionedComparator;
 pub struct KeyRange {
     pub left: Bytes,
     pub right: Bytes,
-    // FIXME: Note that currently `inf` is not always correctly set throughout the codebase,
-    // e.g. there is usage like KeyRange::new(Bytes::new(), Bytes::new()).
-    // It doesn't affect correctness only because we don't use `inf` anywhere for now.
-    pub inf: bool,
 }
 
 impl KeyRange {
```
Comment (Li0k) [inline src/storage/hummock_sdk/src/key_range.rs:27]: shall we use `start_bound_inf` and `end_bound_inf` function to shadow the check logic like `self.right.is_empty()`  which make it more readable

```rust
fn start_bound_inf(&self) -> bool {
  self.left.is_empty()
}
```

**Review by Li0k — APPROVED** [review]
Rest LGTM, thanks

File: src/storage/hummock_sdk/src/key_range.rs (hunk 1 above)
Comment (Little-Wallace) [reply src/storage/hummock_sdk/src/key_range.rs:27]: good idea

**Pull request merged**